tools/rebase-onto-formatter.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+command -v sed  >/dev/null 2>&1 || { echo >&2 "sed is not installed, aborting."; exit 1; }
+
+BASE_COMMIT=$1
+TARGET_COMMIT=$2
+FORMATTING_COMMAND='make formatf'
+USAGE="
+USAGE: $0 BASE_COMMIT TARGET_COMMIT
+    BASE_COMMIT:
+    TARGET_COMMIT:
```

Comment:
```suggestion
    BASE_COMMIT: if X is the commit that changes code formatting (and ideally nothing else), this should be X~1
    TARGET_COMMIT: X
```

So...  isn't it easier to require the formatting happens in one commit, and then just asking for that commit, which gives us both X and X-1?

Reply:
Yes, the easiest case would be if there are two commits, the first one changing the version/settings/whatever and the second one purely applying the formatting.
But it doesn't fit with the way we currently merge code (squashing PRs), so this provides a more general solution. You can provide the doing-everything-at-once commit as TARGET_COMMIT and the only-change-the-settings commit can be passed separately as BASE_COMMIT that does not have to be in the history of `develop`. Usually we can just take it from the pre-squashing history of the PR.

It also doesn't make much of a difference for the complexity of the script.